File: src/frontend/eshop/src/models/products/responses/ProductDetailsResponse.ts
export type ProductDetailsResponse = {
    id: string;
    name: string;
    brand: string;
    description: string;
    mediaUrl?: string;
    price: number;
    categoryId: string;
    brandId: string;
    createdAt: string;
    specifications: Record<string, unknown>;
    availableQuantity: number;
}

File: src/backend/Modules/Catalog/Modules.Catalog/Application/DTOs/Products/ProductDetailsResponse.cs
namespace Modules.Catalog.Application.DTOs.Products;

public sealed record ProductDetailsResponse(
    Guid Id,
    string Name,
    string Brand,
    string Description,
    string? MediaUrl,
    decimal Price,
    Guid CategoryId,
    Guid BrandId,
    DateTime CreatedAt,
    Dictionary<string, object?> Specifications,
    int AvailableQuantity);